overlord/snapstate/backend/copydata.go

```diff
@@ -239,3 +239,91 @@ var removeIfEmpty = func(dir string) error {
 
 	return os.Remove(dir)
 }
+
+var maybeFailForTesting func() error
+
+// InitExposedSnapHome creates and initializes ~/Snap/<snapName> based on the
+// specified revision. Must be called after the snap has been migrated.
+func (b Backend) InitExposedSnapHome(snapName string, rev snap.Revision) (err error) {
+	opts := &dirs.SnapDirOptions{HiddenSnapDataDir: true}
+
+	users, err := allUsers(opts)
+	if err != nil {
+		return err
+	}
+
+	for _, usr := range users {
+		uid, gid, err := osutil.UidGid(usr)
+		if err != nil {
+			return err
+		}
+
+		if maybeFailForTesting != nil {
+			osutil.MustBeTestBinary("'maybeFailForTesting' should only be used in tests")
+			if err := maybeFailForTesting(); err != nil {
+				return err
+			}
+		}
```

Comment:
mmm... I can't say that I like this a lot :-)

Since you are using a couple of methods below which would trigger a similar behaviour (`osutil.MkdirAllChown` and `ioutil.ReadDir`), why not mock them instead?